### PR TITLE
Allow specifying one or more tenants for activate/stream jobs in Java client

### LINKED ISSUE #13563: Add experimental tenantIds methods to activate/stream jobs commands
We must add a new `tenantIds(String...)` methods (taking list, vargargs, etc) to the activate jobs command builder.

The API must be marked as experimental. Implementation should not do anything yet. This will be added later in:
- https://github.com/camunda/zeebe/issues/13560

# Open question
Should we keep the stream jobs command API out of scope?

### PR DESCRIPTION
**Pull request by korthout**
## Description

This is the third in several pull requests to:
- #13517

This pull request adds the ability to specify one or more tenants for the activate jobs and stream jobs command builders.

This pull request contains a new reusable interface to add tenants through several methods:
- `tenantId(String)` -> adds a single tenant
- `tenantIds(List<String>)` -> adds a list of tenants
- `tenantIds(String...)` -> also adds a list of tenants, shorthand for `tenantIds(List<String>)`

Please note that the new methods on the command builders have no effect at this time. The feature is still under development. The API is added so others can already start using this method during development. I'll inform the relevant teams once this is available to them.

## Related issues

closes #13563

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by berkaycanbc — APPROVED** [review]
Another great job! 🚀 

I only left suggestion comments. 👍

File: clients/java/src/main/java/io/camunda/zeebe/client/api/command/CommandWithOneOrMoreTenantsStep.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.api.command;
+
+import io.camunda.zeebe.client.api.ExperimentalApi;
+import java.util.List;
+
+public interface CommandWithOneOrMoreTenantsStep<T> extends CommandWithTenantStep<T> {
```
[the remark this answers is not in the document]
Comment (korthout) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/CommandWithOneOrMoreTenantsStep.java:21]: @berkaycanbc @koevskinikola Is it clear that multiple means one or more? I always feel that multiple means more than one. If you like that better then I can adjust it 👍

[the remark this answers is not in the document]
Comment (koevskinikola) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/CommandWithOneOrMoreTenantsStep.java:21]: @korthout according to [this definition](https://style.mla.org/many-or-multiple/#:~:text=The%20MLA%20follows%20Merriam%2DWebster's,%2C%20Adj.%2C%E2%80%9D%20defs.), the usage of `multiple` here would be incorrect. So I'm fine with your original naming.

**Comment by korthout** [comment]
Ok, just gonna leave it as it is then 👍 

bors merge

**Pull request merged**